examples/version-migration/schema-upgrade/src/view/appView.tsx

```diff
@@ -3,14 +3,17 @@
  * Licensed under the MIT License.
  */
 
+import type { IMigratableModel } from "@fluid-example/example-utils";
 import React, { useEffect, useState } from "react";
 
 import type { IInventoryListAppModel } from "../modelInterfaces.js";
 
 import { InventoryListView } from "./inventoryView.js";
 
 export interface IInventoryListAppViewProps {
-	model: IInventoryListAppModel;
+	// TODO: All we really want here is a "readonly" indicator - maybe don't need the full IMigratableModel interface.
+	// Would maybe be better to grab that info from the Migrator rather than the MigrationTool anyway.
+	model: IInventoryListAppModel & IMigratableModel;
```

Comment:
Couple questions - Do you think this syntax might be confusing for partners to use/is there any value to making an interface that combines the two?

Could you also explain the scenario where customers would want to avoid implementing events?

Reply:
Re: combining - Possibly yeah.  I think I might explore the option in the TODO first though, which would be to explicitly inject the readonly decision separately.  I'd love for customers to be able to design their IInventoryListAppModel-equivalent without having to think about migration, so the separation is nice for that.  The `.version` is still maybe interesting to the customer though (for the purpose of matching model to view, if they want to permit rendering of old models), so I'll need to think about that a bit more.

Re: avoiding implementing events - this change is a good example where the InventoryListAppModel implementations (done by the customer) become simpler because they get to drop the responsibility of echoing the connected event.  In this case there were no other events to emit, so it can drop implementing events entirely.